pandas/tests/io/formats/style/test_html.py

```diff
@@ -467,3 +467,42 @@ def test_maximums(styler_mi, rows, cols):
     assert ">5</td>" in result  # [[0,1], [4,5]] always visible
     assert (">8</td>" in result) is not rows  # first trimmed vertical element
     assert (">2</td>" in result) is not cols  # first trimmed horizontal element
+
+
+def test_omit_css_style_rules_for_hidden_cells():
+    # GH 43619
+    df = DataFrame([[1, 2], [3, 4]], columns=["A", "B"])
+    result = (
+        df.style.set_uuid("")
+        .background_gradient()
+        .hide_columns(["A"])
+        .hide_index([0])
+        .to_html()
+    )
+    expected = dedent(
+        """\
+        <style type="text/css">
+        #T__row1_col1 {
+          background-color: #023858;
+          color: #f1f1f1;
+        }
+        </style>
+        <table id="T_">
```

Comment:
everything below `</style>` is already tested. To minimise and make the test more specific just try `assert expected in result` where `expected = <style>...</style>`

Reply:
done